test/unit/core/error_helpers.js

```diff
@@ -532,23 +532,94 @@ suite('Global Error Handling', function() {
     p5._fesLogger = null;
   });
 
+  testUnMinified('identifies errors happenning internally', function() {
+    return new Promise(function(resolve) {
+      // quite an unusual way to test, but the error listerner doesn't work
+      // under mocha. Also the stacktrace gets filled with mocha internal
+      // function calls. Using this method solves both of these problems.
+      // This method also allows us to test for SyntaxError without messing
+      // with flow of the other tests
+      iframe = createP5Iframe(
+        [
+          P5_SCRIPT_TAG,
+          WAIT_AND_RESOLVE,
+          '<script>',
+          'function setup() {',
+          'let cnv = createCanvas(400, 400);',
+          'cnv.mouseClicked();', // Error in p5 library as no callback passed
+          '}',
+          '</script>'
+        ].join('\n')
+      );
+      log = [];
+      iframe.elt.contentWindow.logger = logger;
+      iframe.elt.contentWindow.afterSetup = resolve;
```

Comment:
I noticed this repeats every test - if you're up to it, here's a way to brush off a couple of lines
1. Make a new function (`prepSyntaxTest()`, for instance)
2. It receives the relevant source code (ie. lines `547` to `549` here) as a list of strings
3. It build the final array by adding `P5_SCRIPT_TAG`, `WAIT_AND_RESOLVE` and the `script` tags
4. it runs this block with it.

This new function can even return the actual Promise for you, so you don't have to retype it for each test case. To do that, the function in the "then" block must become a new parameter.

Reply:
I did the changes please have a look